pandas/core/frame.py

```diff
@@ -3489,33 +3488,30 @@ def _get_info_slice(obj, indexer):
                 )
             )
 
-        # empty include/exclude -> defaults to True
-        # three cases (we've already raised if both are empty)
-        # case 1: empty include, nonempty exclude
-        # we have True, True, ... True for include, same for exclude
-        # in the loop below we get the excluded
-        # and when we call '&' below we get only the excluded
-        # case 2: nonempty include, empty exclude
-        # same as case 1, but with include
-        # case 3: both nonempty
-        # the "union" of the logic of case 1 and case 2:
-        # we get the included and excluded, and return their logical and
-        include_these = Series(not bool(include), index=self.columns)
-        exclude_these = Series(not bool(exclude), index=self.columns)
-
-        def is_dtype_instance_mapper(idx, dtype):
-            return idx, functools.partial(issubclass, dtype.type)
-
-        for idx, f in itertools.starmap(
-            is_dtype_instance_mapper, enumerate(self.dtypes)
-        ):
-            if include:  # checks for the case of empty include or exclude
-                include_these.iloc[idx] = any(map(f, include))
-            if exclude:
-                exclude_these.iloc[idx] = not any(map(f, exclude))
+        # We raise when both include and exclude are empty
+        # Hence, we can just shrink the columns we want to keep
+        keep_these = Series(True, index=self.columns)
```

Comment:
this is significantly more complex and need refactoring

Reply:
Could you elaborate a bit? Do you mean the comment? If so, I intended to keep it short as the code is, in my opinion more clearly in what it does. Previously we had the `exclude_these` array, but from the `and` concatenation at the we keep the columns in there. This didn't make too much sense. I would agree with the `filter_unique_dtypes_on` function though, but I don't see a good way to refactor it. Suggestions welcome.

Reply:
Instead of constructing a Series can you just use a lower level mask using an ndarray?

Reply:
Will implement that. Actually, due to your emphasis on using a mask, I think we can also entirely get rid of  `_get_info_slice` some lines above. We don't need to create slices as we are using a mask.

Reply:
I replaces the series by a boolean Numpy array